src/yaml.cr

```diff
@@ -166,19 +166,20 @@ module YAML
   end
 
   # Checks to see if the value is reserved
-  def self.reserved_value?(value, checks = 0)
+  def self.reserved_value?(value)
     return true if YAML::RESERVED_VALUES.includes?(value)
-    case {value[0]?, value[1]?, value[2]?, value[3]?, value[4]?}
-    when {.try(&.ascii_number?), .try(&.ascii_number?), .try(&.ascii_number?), .try(&.ascii_number?), '-'}
-      (Time::Format::ISO_8601_DATE_TIME.parse(value) rescue false)
-    when {.try(&.ascii_number?), _, _, _, _},
-         {'-', .try(&.ascii_number?), _, _, _},
-         {'+', .try(&.ascii_number?), _, _, _},
-         {'.', .try(&.ascii_number?), _, _, _},
-         {'-', '.', .try(&.ascii_number?), _, _},
-         {'+', '.', .try(&.ascii_number?), _, _}
-      clean_value = value.gsub('_', "")
-      clean_value.to_f64? || clean_value.to_i64?(prefix: true)
-    end
+    reserved = case {value[0]?, value[1]?, value[2]?, value[3]?, value[4]?}
+               when {.try(&.ascii_number?), .try(&.ascii_number?), .try(&.ascii_number?), .try(&.ascii_number?), '-'}
+                 (Time::Format::ISO_8601_DATE_TIME.parse(value) rescue false)
+               when {.try(&.ascii_number?), _, _, _, _},
+                    {'-', .try(&.ascii_number?), _, _, _},
+                    {'+', .try(&.ascii_number?), _, _, _},
+                    {'.', .try(&.ascii_number?), _, _, _},
+                    {'-', '.', .try(&.ascii_number?), _, _},
+                    {'+', '.', .try(&.ascii_number?), _, _}
+                 clean_value = value.gsub('_', "")
+                 clean_value.to_f64? || clean_value.to_i64?(prefix: true)
+               end
+    !reserved.nil?
```

Comment:
Why not coerce this inside the case, it will likely look cleaner.

Reply:
Done